resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -867,6 +868,20 @@ object ApplicationMaster extends Logging {
         val originalCreds = UserGroupInformation.getCurrentUser().getCredentials()
         SparkHadoopUtil.get.loginUserFromKeytab(principal, sparkConf.get(KEYTAB).orNull)
         val newUGI = UserGroupInformation.getCurrentUser()
+
+        if (master.isClusterMode) {
```

Comment:
the check above is already checking cluster mode only (amArgs.userClass != null) though that would be much more readable if replaced with this.

Reply:
Ah right honestly I missed `amArgs.userClass != null` in pattern matching. Thanks for the pointer!